tests/test_package_manager/test_v1.py

```diff
@@ -0,0 +1,234 @@
+                "Both `update_packages` and `update_hashes` cannot be set to `True`."
+            ),
+        ):
+            pm.sync(update_hashes=True, update_packages=True)
+
+        with mock.patch.object(pm, "add_package") as update_patch:
+            pm.sync(dev=True, third_party=False)
+            update_patch.assert_called_with(package_id=DUMMY_PACKAGE_ID)
+
+        pm = PackageManagerV1(
+            path=self.packages_dir_path,
+            third_party_packages=OrderedDict({DUMMY_PACKAGE_ID: DUMMY_PACKAGE_HASH}),
+        )
+
+        with mock.patch.object(pm, "add_package") as update_patch:
+            pm.sync(dev=False, third_party=True)
+            update_patch.assert_called_with(package_id=DUMMY_PACKAGE_ID)
+
+
+class TestHashUpdateDev(BaseAEATestCase):
+    """Test hash update."""
+
+    use_packages_dir: bool = True
+
+    def test_update_package_hashes(
+        self,
+    ) -> None:
+        """Test update package hash method."""
+
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+
+        original_hash = packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path]
+        packages_v1["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+
+        packages_json_file = self.packages_dir_path / "packages.json"
+        packages_json_file.write_text(json.dumps(obj=packages_v1))
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+
+        pm.update_package_hashes().dump()
+
+        packages_v1_updated = json.loads(packages_json_file.read_text(encoding="utf-8"))
+
+        assert pm.dev_packages[EXAMPLE_PACKAGE_ID] == original_hash
+        assert (
+            packages_v1_updated["dev"][EXAMPLE_PACKAGE_ID.to_uri_path] == original_hash
+        )
+
+
+class TestHashUpdateThirdParty(BaseAEATestCase):
+    """Test hash update."""
+
+    use_packages_dir: bool = True
+
+    def test_update_package_hashes(self, caplog) -> None:
+        """Test update package hash method."""
+
+        packages_v1 = json.loads(PACKAGE_JSON_FILE.read_text(encoding="utf-8"))
+        original_hash = packages_v1["dev"].pop(EXAMPLE_PACKAGE_ID.to_uri_path)
+        packages_v1["third_party"][EXAMPLE_PACKAGE_ID.to_uri_path] = DUMMY_PACKAGE_HASH
+
+        packages_json_file = self.packages_dir_path / "packages.json"
+        packages_json_file.write_text(json.dumps(obj=packages_v1))
+        pm = PackageManagerV1.from_dir(self.packages_dir_path)
+
+        with caplog.at_level(logging.WARNING):
+            pm.update_package_hashes().dump()
+            packages_v1_updated = json.loads(
+                packages_json_file.read_text(encoding="utf-8")
+            )
+
+            assert pm.third_party_packages[EXAMPLE_PACKAGE_ID] == original_hash
+            assert (
+                packages_v1_updated["third_party"][EXAMPLE_PACKAGE_ID.to_uri_path]
+                == original_hash
+            )
+            assert "Hash change detected for third party package" in caplog.text
```

Comment:
This does not exist for `v0`, why? Because the `get_package_hash` implementation is different there!
https://github.com/valory-xyz/open-aea/blob/46cbcc4e6eebc15947e418e75e1d493c3c3e8eac/aea/package_manager/v0.py#L62-L65
https://github.com/valory-xyz/open-aea/blob/46cbcc4e6eebc15947e418e75e1d493c3c3e8eac/aea/package_manager/v1.py#L80-L85
which affects this piece of code of your PR where you introduce the feature
https://github.com/valory-xyz/open-aea/blob/fd0e7b06f54386a30b9a43b650e0fd7f8916dccf/aea/package_manager/base.py#L169-L172

Reply:
[fix: use PackageId.without_hash() when retrieving package hash from…](https://github.com/valory-xyz/open-aea/pull/464/commits/2db8805258b39fee6a346c7d867886f38ac7abc8) addressed